qiskit/circuit/quantumcircuit.py

```diff
@@ -1941,6 +1961,29 @@ def _bind_parameter(self, parameter, value):
                 parameter in self.global_phase.parameters):
             self.global_phase = self.global_phase.bind({parameter: value})
 
+    def _bind_all_parameters(self, param_dict):
+        """Assigns parameter values to matching instructions in-place."""
```

Comment:
Can you add a note explaining how this function differs from looping `_bind_parameter` over parameters?